Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/BufferPoolDescriptor.h

```diff
@@ -23,6 +23,7 @@ namespace AZ
         public:
             virtual ~BufferPoolDescriptor() = default;
             AZ_RTTI(BufferPoolDescriptor, "{53074556-41D1-4246-8BF0-E5B096997C65}", ResourcePoolDescriptor);
+            AZ_CLASS_ALLOCATOR(BufferPoolDescriptor, SystemAllocator, 0);
```

Comment:
nit: Can this be moved to above AZ_RTTI? So far its been consistently declared before AZ_RTTI (if any) to this point. There are also a few others in the PR that are not in this order.

Reply:
Fixed.